File: mooc-2024/part-4/4-define-iteration/10-the-length-of-the-longest-in-the-list/main.c
#include <stdio.h>
#include <string.h>

int main() {
    int len = 4;
    int max_len = 0;
    char arr[][50] = {"first", "second", "fourth", "eleventh"};
    
    for (int idx = 0; idx < len; idx++) {
        if (strlen(arr[idx]) > max_len) {
            max_len = strlen(arr[idx]);
        }
    }

    printf("The maximum length: %d\n", max_len);

    return 0;
}

File: mooc-2024/part-4/4-define-iteration/10-the-length-of-the-longest-in-the-list/main.py
def length_of_longest(l):
    return max(map(len, l))

my_list = ["first", "second", "fourth", "eleventh"]

result = length_of_longest(my_list)
print(result)
my_list = ["adele", "mark", "dorothy", "tim", "hedy", "richard"]

result = length_of_longest(my_list)
print(result)